packages/ripple-nuxt-tide/modules/preview/lib/helpers.js

```diff
@@ -0,0 +1,93 @@
+import { redirect } from '../config/paths'
+import jsonapiParse from 'jsonapi-parse'
+
+async function tidePreview (context, headersConfig) {
+  if (!context.$auth.loggedIn) {
+    return context.redirect(`${redirect.login}?destination=${context.req.url}`)
+  } else {
+    return getPageByPreviewLink(context.app.$tide, context.route.path, context.store.state.tide.siteId, context.route.query.section, headersConfig)
+  }
+}
+
+async function tidePreviewLogin ($route, $auth, storage) {
+  const isLoginInProgress = ($route.query.code !== undefined)
+  if (!$auth.loggedIn && !isLoginInProgress) {
+    // Store destination.
+    if ($route.query.destination) {
+      storage.setItem('login-destination', $route.query.destination)
+    }
+    await $auth.loginWith('drupal')
+  }
+}
+
+function tidePreviewSuccessRedirect ($router, storage) {
+  const destination = storage.getItem('login-destination')
+  if (destination) {
+    // Go to destination.
+    storage.removeItem('login-destination')
+    $router.replace({ path: destination })
+  } else {
+    // Go to home.
+    $router.replace({ path: '/' })
+  }
+}
+
+async function getPageByPreviewLink ($tide, path, site, section, headersConfig) {
+  const params = {}
+  const { 2: contentType, 3: uuid, 4: revisionId } = path.split('/')
+
+  params.resourceVersion = (revisionId === 'latest') ? 'rel:working-copy' : `id:${revisionId}`
+
+  const pathData = {
+    entity_type: 'node',
+    bundle: contentType,
+    uuid: uuid
+  }
+  const entity = await $tide.getEntityByPathData(pathData, params, headersConfig)
+  if (entity instanceof Error) {
+    throw entity
+  }
+  const pageData = jsonapiParse.parse(entity).data
+
+  // Append the site section to page data
+  pageData.section = (section && section !== site) ? section : null
+  return pageData
+}
+
+async function tideShare (context, headersConfig) {
+  return getPageByShareLink(context.app.$tide, context.route.path, context.store.state.tide.siteId, context.route.query.section, headersConfig)
```

Comment:
Same site id is not section id here.

Reply:
As per the above getPageByPreviewLink() discussion, this implementation should be what you're after.